app/binary-intro/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import React, { useState } from 'react';
+import { Typography, Stack, Button, Box } from '@mui/material';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
+import AutorenewIcon from '@mui/icons-material/Autorenew';
+
+const Page = () => {
+  const [decimalNumber, setDecimalNumber] = useState(0);
+
+  const increment = () => {
+    setDecimalNumber(decimalNumber + 1);
+  };
+
+  const decrement = () => {
+    if (decimalNumber > 0) {
+      setDecimalNumber(decimalNumber - 1);
+    }
+  };
+
+  const reset = () => {
+    setDecimalNumber(0);
+  };
```

Comment:
This can probably be shortened to:

```tsx
  const increment = () => setDecimalNumber(n => n + 1);
  const decrement = () => setDecimalNumber(n => Math.max(0, n - 1));
  const reset = () => setDecimalNumber(0);
  ```

Reply:
ok